client/contentrouting.go

```diff
@@ -0,0 +1,54 @@
+package client
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/ipfs/go-cid"
+	"github.com/libp2p/go-libp2p-core/peer"
+)
+
+type ContentRoutingClient struct {
```

Comment:
Go certainly doesn't require this but if you want to assert interface compliance with one or more of the routing interfaces from https://github.com/libp2p/go-libp2p-core/blob/525a0b13017263bde889a3295fa2e4212d7af8c5/routing/routing.go#L67 then you can do so with the pattern `var _ MyInterface = (*myStruct)(nil)`.

This means you won't need accidentally break an interface without realizing it.